### ISSUE TITLE
influx_tsm panic when trying to convert data from 0.9.6

### ISSUE THREAD

**Issue by rvrignaud** [influxdata/influxdb#5374]
Hello,

I'm trying to convert data from 0.9.6.1 with influx_tsm.
Influxdb was installed with debian package "influxdb_0.10.0-0.beta1_amd64.deb" on Ubuntu 14.04.
Here are the results:

```
# influx_tsm  /opt/influxdb/data

b1 and bz1 shard conversion.
-----------------------------------
Data directory is:        /opt/influxdb/data
Databases specified:      all
Database backups enabled: yes
Parallel mode enabled:    no

unexpected fault address 0x7f28cb9c300a
fatal error: fault
[signal 0xb code=0x2 addr=0x7f28cb9c300a pc=0x5b0faa]

goroutine 1 [running]:
runtime.gothrow(0x97bee0, 0x5)
        /usr/local/go/src/runtime/panic.go:503 +0x8e fp=0xc2080b35c8 sp=0xc2080b35b0
runtime.sigpanic()
        /usr/local/go/src/runtime/sigpanic_unix.go:29 +0x265 fp=0xc2080b3618 sp=0xc2080b35c8
github.com/boltdb/bolt.(*freelist).read(0xc2080c25d0, 0x7f28cb9c3000)
        /tmp/tmp.TcaAgGugKM/src/github.com/boltdb/bolt/freelist.go:162 +0x2a fp=0xc2080b36b0 sp=0xc2080b3618
github.com/boltdb/bolt.Open(0xc2080c23f0, 0x2e, 0x1b6, 0xc2080b3838, 0x0, 0x0, 0x0)
        /tmp/tmp.TcaAgGugKM/src/github.com/boltdb/bolt/db.go:216 +0x673 fp=0xc2080b37f8 sp=0xc2080b36b0
github.com/influxdb/influxdb/cmd/influx_tsm/tsdb.shardFormat(0xc2080c23f0, 0x2e, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/cmd/influx_tsm/tsdb/database.go:209 +0x23c fp=0xc2080b38a0 sp=0xc2080b37f8
github.com/influxdb/influxdb/cmd/influx_tsm/tsdb.(*Database).Shards(0xc20802b010, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/cmd/influx_tsm/tsdb/database.go:172 +0x545 fp=0xc2080b3ab8 sp=0xc2080b38a0
main.main()
        /tmp/tmp.TcaAgGugKM/src/github.com/influxdb/influxdb/cmd/influx_tsm/main.go:113 +0x373f fp=0xc2080b3f98 sp=0xc2080b3ab8
runtime.main()
        /usr/local/go/src/runtime/proc.go:63 +0xf3 fp=0xc2080b3fe0 sp=0xc2080b3f98
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2232 +0x1 fp=0xc2080b3fe8 sp=0xc2080b3fe0

goroutine 6 [syscall]:
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
        /usr/local/go/src/os/signal/signal_unix.go:27 +0x35
```

**Comment by rvrignaud** [comment]
After deleting one of my DB, I was able to migrate data.
It took lots of hours on a 2 core VM with 8Gb of RAM (could not use parallel as it was killed by OOM).
The compression benefit is impressive.

**Comment by jwilder** [comment]
@benbjohnson Any idea what the bolt panic might be?

@rvrignaud What version of InfluxDB were your shards created with? Also, what storage engine were you using before?

**Comment by benbjohnson** [comment]
@jwilder It looks like a partially written Bolt file. It's opening the bolt file and the meta page validation checks out OK but when it goes to get the freelist page it is reaching beyond the size of the file and hitting a page fault.

@rvrignaud Were the data files copied before conversion?

**Comment by rvrignaud** [comment]
@jwilder : I have now way to know which version created the deleted database. I used nearly all released version from pre 0.9 release to 0.10.0-beta.

@benbjohnson no the data were never moved

**Comment by benbjohnson** [comment]
@mark-rushakoff I [added a `bolt rollback` command](https://github.com/benbjohnson/bolt/tree/rollback) that will revert Bolt to the previous commit:

```sh
$ cd $GOPATH/src/github.com/boltdb/bolt
$ git remote add benbjohnson https://github.com/benbjohnson/bolt
$ git fetch benbjohnson
$ git checkout rollback
$ go install ./cmd/bolt
$ bolt rollback /path/to/my/data/file
```

I'm not sure why the data file would be getting truncated. I haven't seen this issue on Bolt anywhere but with this conversion tool.
